CONTRIBUTING.md

```diff
@@ -627,6 +631,74 @@ required.
 Essentially, the steps you have to take to deploy manually are the same that
 `skaffold` will perform, and they're outlined at the previous section.
 
+#### Kubernetes E2E tests
+
+Kubernetes integration is quite complicated, a lot of things can go wrong.
+
+To cope with the complexity and ensure we maintain high quality, we use
+E2E (end-to-end) tests.
+
+> E2E tests normally run at CI, so there's typically no need to run them
+> manually.
+
+##### Requirements
+
+* `kubernetes` cluster (`minikube` has special support, but any cluster should
+  work)
+* `docker`
+* `kubectl`
+* `bash`
+
+Vector release artifacts are prepared for E2E tests, so the ability to do that
+is required too, see Vector [docs](https://vector.dev) for more details.
+
+##### Running the E2E tests
+
+To run the E2E tests, use the following command:
+
+```shell
+CONTAINER_IMAGE_REPO=<your name>/vector-test make test-e2e-kubernetes
+```
+
+Where `CONTAINER_IMAGE_REPO` is the docker image repo name to use, without part
+after the `:`. Replace `<your name>` with your Docker Hub username.
+
+This will run Kubernetes E2E tests suite. If the command exit with 0 status
+code - tests passed.
+
+You can also pass additional parameters to adjust the behavior of the test:
+
+* `QUICK_BUILD=true` - use development build and a skaffold image from the dev
+  flow instead of a production docker image. Significantly speeds up the
+  preparation process, but doesn't guarantee the correctness in the release
+  build. Useful for development of the tests or Vector code to speed up the
+  iteration cycles.
+
+* `USE_MINIKUBE_CACHE=true` - instead of pushing the built docker image to the
+  registry under the specified name, directly load the image into
+  a `minikube`-controlled cluster node.
+  Requires you to test against a `minikube` cluster. Eliminates the need to have
+  a registry to run tests.
+
+* `CONTAINER_IMAGE_REPO=<your name>/vector-test:tag` - completely skip the step
```

Comment:
I guess this exists because there is no `make build-docker-....` images and the `release` tasks are kind of awkward?

Reply:
Oops, there's a mistake. This is supposed to be `CONTAINER_IMAGE`, not `CONTAINER_IMAGE_REPO`. Hope it makes more sense.
This can be set if you want to bypass image building altogether, and use a completely custom image. This might be useful, for instance, for testing release images. I used it for experimenting with vector building, and to avoid rebuilding vector each time I change tests. Overall, it's very useful to have such option. Sorry for an error - this is definitely confusing as is.

Reply:
Also, if the `CONTAINER_IMAGE` is not set - a new image is built using the `CONTAINER_IMAGE_REPO` and a unique random tag. This process uses `make` (unless `QUICK_BUILD` is requested).